qemu/tests/slof_user_interface.py

```diff
@@ -0,0 +1,176 @@
+            return ''.join(content)
+        return None
+
+    def _check_menu_info(menu_info):
+        """ Check the menu info by each items. """
+        bootable_num = ''
+        for i in range(1, int(params['boot_dev_num']) + 1):
+            option = params['menu_option%d' % i]
+            logging.info('Checking the device(%s) if is included in menu list.'
+                         % '->'.join(option.split()))
+
+            dev_type, hba_type, child_bus, addr = option.split()
+            addr = re.sub(r'^0x0?', '', addr)
+            pattern = re.compile(r'(\d+)\)\s+%s(\d+)?\s+:\s+/%s(\S+)?/%s@%s' %
+                                 (dev_type, hba_type, child_bus, addr), re.M)
+            searched = pattern.search(menu_info)
+            if not searched:
+                test.fail('No such item(%s) in boot menu list.' %
+                          '->'.join(option.split()))
+            if i == int(params['bootable_index']):
+                bootable_num = searched.group(1)
+        return bootable_num
+
+    def _enter_user_interface(mode):
+        """ Enter user interface. """
+        o = utils_misc.wait_for(
+            lambda: _send_key(enter_key[mode], False), ack_timeout, step=0.0)
+        if not o:
+            test.fail('Failed to enter user interface in %s sec.' % ack_timeout)
+        return o
+
+    def _f12_user_interface_test():
+        """ Test f12 user interface. """
+        menu_list = _enter_user_interface(F12)
+        actual_num = len(re.findall(r'\d+\)', menu_list))
+        dev_num = params['boot_dev_num']
+        if actual_num != int(dev_num):
+            test.fail(
+                'The number of boot devices is not %s in menu list.' % dev_num)
+        if not utils_misc.wait_for(
+                lambda: _send_key(
+                    _check_menu_info(menu_list), False), ack_timeout, step=0.0):
+            test.fail('Failed to load after selecting boot device '
+                      'in %s sec.' % ack_timeout)
+
+    def _load_user_interface_test():
+        """ Test boot/reset-all user interface. """
+        _enter_user_interface(STOP)
+        if not utils_misc.wait_for(
+                lambda: _send_key(keys, True, 3), ack_timeout, step=0.0):
+            test.fail(
+                'Failed to load after \'%s\' in %s sec.' % (keys, ack_timeout))
+
+    def _check_serial_log_status():
+        """ Check the status of serial log. """
+        file_timeout = 30
+        if not utils_misc.wait_for(
+                lambda: os.path.isfile(vm.serial_console_log),
+                file_timeout):
+            test.error('No found serial log during %s sec.' % file_timeout)
+
+    main_tests = {'f12':  _f12_user_interface_test,
+                  'boot': _load_user_interface_test,
+                  'reset-all': _load_user_interface_test}
+
+    try:
+        create_cd = params.get('create_cd', 'no')
+        if create_cd == 'yes':
+            _create_iso()
+
+        params['start_vm'] = 'yes'
```

Comment:
I think you can remove `params['start_vm'] = 'yes'`

Reply:
thanks, have updated.